File: abc/abc327/b.java
// https://atcoder.jp/contests/abc327/submissions/47345430

import java.util.Scanner;

class Main {
  static long pow(long a, long b) {
    long res = 1;
    while(b > 0) {
      b--;
      res *= a;
    }

    return res;
  }

  public static void main(String args[]) {
    Scanner stdIn = new Scanner(System.in);

    long b = stdIn.nextLong();
    long max = 10000000;

    for(long i = 1; i < max; i++) {
      if(pow(i, i) > b) break;
      if(pow(i, i) == b) {
        System.out.print(i);
        System.exit(0);
      }
    }

    System.out.print(-1);
  }
}

File: abc/abc327/b.rs
// https://atcoder.jp/contests/abc327/submissions/47516293

use proconio::input;

fn main() {
    input! {
        b: i128
    }

    let mut ans = -1;
    for i in 1..20 {
        let mut a = 1;
        for j in 0..i {
            a *= i;
        }

        if a == b {
            ans = i;
        }
    }

    print!("{}", ans);
}

